xdsl/dialects/builtin.py

```diff
@@ -373,12 +373,6 @@ def from_int_and_width(value: int, width: int) -> IntegerAttr[IntegerType]:
     def from_index_int_value(value: int) -> IntegerAttr[IndexType]:
         return IntegerAttr(value, IndexType())
 
-    @staticmethod
-    def from_params(
-        value: int | IntAttr, typ: int | IntegerType | IndexType
-    ) -> IntegerAttr[IntegerType | IndexType]:
-        return IntegerAttr(value, typ)
```

Comment:
Weirdly, pyright now complains about passing `int` as second parameter, but only here and in `riscv.py`, and not in other places we do that in the code. I thought I'd just remove this method since it's redundant, and pass `i32` in `riscv.py` for now

Reply:
Yes let's remove it, this method should be replaced by the `__init__` anyway.
The reason why `pyright` complains is because of the `@overload` in `IntegerAttr.__init__` I believe.